server/lib/puppet-languageserver/puppet_parser_helper.rb

```diff
@@ -53,14 +66,23 @@ def self.object_under_cursor(content, line_num, char_num, multiple_attempts = fa
 
       result = nil
       move_offset = 0
-      %i[noop remove_char try_quotes try_quotes_and_comma].each do |method|
+      %i[noop remove_word try_quotes try_quotes_and_comma remove_char].each do |method|
         new_content = nil
         case method
         when :noop
           new_content = content
         when :remove_char
           new_content = remove_char_at(content, line_offsets, line_num, char_num)
           move_offset = -1
+        when :remove_word
+          next_char = get_char_at(content, line_offsets, line_num, char_num)
+
+          while /\w/.match(next_char)
```

Comment:
I'll switch it to `/[[:word:]]/` which will match ["a character in one of the following Unicode general categories Letter, Mark, Number, Connector_Punctuation"](http://ruby-doc.org/core-2.3.1/Regexp.html)